stdlib/asm/crypto/fri/frie2f4.masm

```diff
@@ -262,82 +248,52 @@ end
 #!
 #! The memory referenced above is used contiguously, as follows:
 #!
-#!   [layer_ptr ... rem_ptr ... query_start_ptr ... query_end_ptr]
+#!   [query_ptr ... layer_ptr ... rem_ptr ...]
 #!
 #! This means for example that:
 #! 1. rem_ptr - 1 points to the last (alpha0, alpha1, t_depth, d_size) tuple.
-#! 2. The length of the remainder codeword is 2 * (rem_ptr - query_start_ptr).
+#! 2. layer_ptr - 1 points to the last (e0, e1, p, poe) tuple.
 #!
 #! Cycles: for domains of size 2^n where:
-#! - n is even: 12 + 6 + num_queries * (40 + num_layers * 76 + 69) + 2626
-#! - n is odd:  12 + 6 + num_queries * (40 + num_layers * 76 + 69) + 1356
+#! - n is even: 6 + 7 + num_queries * (40 + num_layers * 76 + 26)
+#! - n is odd:  6 + 7 + num_queries * (40 + num_layers * 76 + 26)
 export.verify.1
 
-    # store [query_start_ptr, query_end_ptr, layer_ptr, rem_ptr] to keep track of all queries
+    # store [query_ptr, layer_ptr, rem_ptr, g] to keep track of all queries
     # (3 cycles)
     locaddr.0
     mem_storew
 
-    # compute size of remainder divided by 2 to be used in determining the appropriate remainder
-    # verification procedure. (5 cycles)
+    # [(query_ptr == layer_ptr), query_ptr, layer_ptr, rem_ptr, g]
+    # (3 cycles)
     dup
-    dup.4
-    sub
-    movdn.5
-    # => [query_start_ptr, query_end_ptr, layer_ptr, rem_ptr, g, rem_len]
-
-    # compute number of remaining queries to be verified and assert it is not 0
-    # (4 cycles)
-    swap
-    dup.1
+    dup.2
     neq
-    # => [?, query_start_ptr, layer_ptr, rem_ptr, g, rem_len, ...]
 
     while.true
-        # load [e0, e1, p, 0] from memory i.e. next query data (7 cycles)
+        # load [e0, e1, p, poe] from memory i.e. next query data (7 cycles)
         push.0.0.0.0
         movup.4
         mem_loadw
-        drop
-        # => [p, e1, e0, layer_ptr, rem_ptr, g, rem_len, ...]
-
-        # compute poe = g^p (43 cycles)
-        dup.5
-        dup.1
-        exp.u32
-        # => [poe, p, e1, e0, layer_ptr, rem_ptr, rem_len, ...]
+        # => [poe, p, e1, e0, layer_ptr, rem_ptr, g, ...]
 
         # we now have everything to verify query p
         exec.verify_query
 
-        # prepare for next iteration (17 cycles)
-        # => [x, x, x, x, x, x, x, x, x, x, g, rem_len, ...]
-        dropw drop drop
+        # prepare for next iteration (18 cycles)
+        # => [x, x, x, x, x, x, x, x, x, x, g, ...]
+        dropw drop drop drop
         locaddr.0
-        mem_loadw   # load [query_start_ptr, query_end_ptr, layer_ptr, rem_ptr]
+        mem_loadw   # load [query_ptr, layer_ptr, rem_ptr, g]
```

Comment:
Similarly, we should be able to replace this with:
```
loc_loadw.0
```

Reply:
Done!